src/synonyms/synonymconflation.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+
+"""
+conflate-synonyms.py is a Python script for conflating synonym files using one or more
+conflation files.
+"""
+import sys
+import json
+import logging
+from collections import defaultdict
+
+import click
+
+# Set up default logging.
+logging.basicConfig(level=logging.INFO)
+
+
+#click.command()
+#click.option('--conflation-file', multiple=True, type=click.Path(exists=True))
+#click.option('--output', type=click.Path(exists=False), default='-')
+#click.argument("synonym_files", nargs=-1, type=click.Path(exists=True))
+def conflate_synonyms(synonym_files, conflation_file, output):
+    """
+    Generate a synonym file based on a single input synonym, the conflation described in the input conflation files,
+    and any cross-references present in the input compendia files.
+
+    :param synonym_files: The input synonym files.
+    :param conflation_file: Any conflation files to apply.
+    :param output: The file to write the synonyms to.
+    :return:
+    """
+
+    logging.info(f"conflate_synonyms({synonym_files}, {conflation_file}, {output})")
+
+    # Some common code to manage the conflation index.
+    # This is simply a large dictionary, where every key is an identifier and the value is the identifier to map it to.
+    conflation_index = dict()
+    conflations = defaultdict(list)
+
+    def add_conflation(primary_id, secondary_id):
```

Comment:
It's legal but functions defined in functions always freaks me out.

Reply:
For bonus confusion, this function is only used in one place :). Removed in d15b79b.